### PR TITLE
remove_uses_of_pointer_stringify

### PR DESCRIPTION
**Pull request by juj**
Remove uses of Pointer_stringify() in favor of UTF8ToString(). We can eventually deprecate Pointer_stringify() in favor of UTF8ToString(), to clean up code size.

### CODE REVIEW

File: src/library_browser.js [hunk 1]
```diff
@@ -1483,9 +1483,7 @@ var LibraryBrowser = {
   emscripten_get_preloaded_image_data__proxy: 'sync',
   emscripten_get_preloaded_image_data__sig: 'iiii',
   emscripten_get_preloaded_image_data: function(path, w, h) {
-    if (typeof path === "number") {
-      path = Pointer_stringify(path);
-    }
+    if (path | 0 === path) path = UTF8ToString(path);
```
[the remark this answers is not in the document]
Comment (juj) [reply src/library_browser.js]: Ops, you are completely right!

**Comment by juj** [comment]
> If `stringToNewUTF8` were in preamble.js, the optimizer would still be able to remove it if not used, and (1) can avoid the `_` prefix, (2) can use it from preamble, postamble, etc. I suspect this would be useful in more places, but maybe I'm wrong? No strong feelings either way.

I really hate the fact that I went and put all the UTF8ToString and stringToUTF8 functions into preamble.js, I'd much more like doing this library method instead, so that even if optimizers are not being used, the function will be DCEd away. First order of business is to get good closure-minified and DCEd code, but there is also value to want to favor constructs that also give good debug mode builds out of the box.

I could rename the function to `$stringToNewUTF8` so that it will avoid the `_`, that might make it a little bit easier?

> Pointer_stringify differs from UTF8ToString, in that it also checks if there is any utf8, and uses a faster code path otherwise. Do you think that does not matter anymore?

The fact that `Pointer_stringify` checks if there is utf8 or not feels wishful performance optimization. Instead callers should know what kind of strings they are operating on - if it's utf8, then use `UTF8ToString`, and if it's 7-bit ASCII, then use `AsciiToString`. `UTF8ToString` is definitely faster than `Pointer_stringify` when the source string is UTF-8 (because `Pointer_stringify` first has to check, but then has to do what `UTF8ToString` does), and `AsciiToString` is definitely faster than `Pointer_stringify` for ASCII inputs (because `AsciiToString` does not need to do a redundant scan through the ASCII string that `Pointer_stringify` does).

Only time when `Pointer_stringify` might have an opportunity to be faster than `UTF8ToString` is if one is converting a UTF-8 string that might contain non-ASCII UTF-8 characters, but in actuality does not. Though it is not certain if even then that will be the case, because of the "got to scan the string twice" behavior.

In any either, knowing the encoding of the data trumps, and it's good to have functions that statically deal with the expected encoding, and highlight the encoding in the string conversion function, so that it is clear to user what the conversion function will do. If user is right about what kind of content their string will have, then `UTF8ToString` and `AsciiToString` will be the fastest ways to convert.

**Comment by kripken** [comment]
Makes sense, lgtm. Yeah, I think it'd be nicer with `$stringToNewUTF8`.

**Comment by juj** [comment]
Ok, pushed with `$stringToNewUTF8`, I'll land once CI looks good.

**Pull request merged**